reVX/offshore/offshore_inputs.py

```diff
@@ -0,0 +1,300 @@
+
+    @staticmethod
+    def _parse_offshore_sites(offshore_sites):
+        """
+        Load offshore sites from disc if needed
+
+        Parameters
+        ----------
+        offshore_sites : str | list | tuple | ndarray |pandas.DataFrame
+            - Path to .csv|.json file with offshore sites meta data
+            - Path to a WIND Toolkit .h5 file to extact site meta from
+            - List, tuple, or vector of offshore gids
+            - Pre-extracted site meta DataFrame
+
+        Returns
+        -------
+        offshore_sites : pandas.DataFrame
+            Offshore sites meta data
+        """
+        if isinstance(offshore_sites, str):
+            if offshore_sites.endswith('.h5'):
+                with Resource(offshore_sites) as f:
+                    offshore_sites = f.meta.reset_index()
+            else:
+                offshore_sites = parse_table(offshore_sites)
+        elif isinstance(offshore_sites, (tuple, list, np.ndarray)):
+            offshore_sites = pd.DataFrame({'gid': offshore_sites})
+
+        if not isinstance(offshore_sites, pd.DataFrame):
+            msg = ("offshore sites must be a .csv, .json, or .h5 file path, "
+                   "or a pre-extracted pandas DataFrame, but {} was provided"
+                   .format(offshore_sites))
+            logger.error(msg)
+            raise ValueError(msg)
+
+        if 'offshore' in offshore_sites:
+            mask = offshore_sites['offshore'] == 1
+            offshore_sites = offshore_sites.loc[mask]
+
+        return offshore_sites
+
+    def _create_offshore_meta(self, offshore_sites, tm_dset='techmap_wtk'):
+        """
+        Create offshore meta from offshore sites and techmap
+
+        Parameters
+        ----------
+        offshore_sites : str | pandas.DataFrame
+            Path to .csv file with offshore sites or offshore meta, or path
+            to a .h5 file to extact site meta from, or pre-extracted site meta
+            DataFrame
+        tm_dset : str, optional
+            Dataset / layer name for wind toolkit techmap,
+            by default 'techmap_wtk'
+
+        Returns
+        -------
+        offshore_meta : pandas.DataFrame
+            Offshore sites meta data including mapping to input layers
+        """
+        offshore_sites = self._parse_offshore_sites(offshore_sites)
+        if 'gid' not in offshore_sites:
+            msg = ('Cannot find offshore WIND Toolkit "gid"s of interest! '
+                   'Offshore sites input must have a "gid" column: {}'
+                   .format(list(offshore_sites.columns)))
+            logger.error(msg)
+            raise RuntimeError(msg)
+
+        offshore_gids = offshore_sites['gid'].values
+        tech_map = self._reduce_tech_map(self._inputs_fpath, tm_dset=tm_dset,
+                                         offshore_gids=offshore_gids)
+
+        offshore_meta = pd.merge(offshore_sites, tech_map, on='gid',
+                                 how='inner')
```

Comment:
inner seems dangerous here. Could we unknowingly drop offshore sites? Maybe a simple assert checking that the offshore_meta output has the same length as the offshore_sites input?

Reply:
This was sort of a result of our test data. It turns out that the offshore exclusions don't reach as far out as the resource data, so some of the offshore sites weren't in the techmap. I added this warning:
https://github.com/NREL/reVX/blob/5ac8db1c09519d29b835002f90a3f949420f12ca/reVX/offshore/offshore_inputs.py#L145-L152